### PR TITLE
MM-21364 - Plugin Settings Config Best Practices

### PR DESCRIPTION
**Pull request by marianunez**
#### Summary

Extended documentation on how to define a plugins configuration settings. I've included as a previous question to the custom setting in Extend/Plugins/Best Practices

#### Ticket Link
[MM-21364](https://mattermost.atlassian.net/browse/MM-21364)

### CODE REVIEW

File: site/content/extend/plugins/best-practices.md [hunk 1]
```diff
@@ -6,11 +6,15 @@ weight: 50
 
 See here for [server-specific best practices for plugins](/extend/plugins/server/best-practices/). Webapp-specific best practices are incoming.
 
-## How can I create my own plugin setting?
+## How do I allow my plugin to be configured through the System Console?
```
Comment (justinegeffen) [inline site/content/extend/plugins/best-practices.md]:
```suggestion
## How do I enable plugin configuration through the System Console?
```

File: site/content/extend/plugins/best-practices.md [hunk 2]
```diff
@@ -6,11 +6,15 @@ weight: 50
...
 
-A plugin's admin settings are defined in the plugin's manifest [setting schema](https://developers.mattermost.com/extend/plugins/manifest-reference/#settings_schema). This includes any pre-defined types that the System Console UI currently supports. 
+Once a plugin is installed, Administrators have access to the plugin's configuration page in the `System Console > Plugins` section. The settings that can be configured there must be defined in the plugin's manifest [setting schema](https://developers.mattermost.com/extend/plugins/manifest-reference/#settings_schema). The Web App supports several basic pre-defined settings type, i.e. `bool` and `dropwown`, for which it will provide the corresponding UI components to be able to properly configured in the System Console.
```
Comment (justinegeffen) [inline site/content/extend/plugins/best-practices.md]:
```suggestion
Once a plugin is installed, Administrators have access to the plugin's configuration page in the **System Console > Plugins** section. The configurable settings must first be defined in the plugin's manifest [setting schema](https://developers.mattermost.com/extend/plugins/manifest-reference/#settings_schema). The web app supports several basic pre-defined settings type, e.g. `bool` and `dropdown`, for which the corresponding UI components are provided in order to complete configuration in the System Console.
```

File: site/content/extend/plugins/best-practices.md [hunk 3]
```diff
@@ -6,11 +6,15 @@ weight: 50
...
 
-Additionally, a plugin could define their own type of setting with a corresponding custom user interface that will be displayed in the system console following the process below. 
+These settings are stored within the server configuration under [`Plugins`] indexed by plugin ids. The plugin's server code will be able to access their current configuration calling the [`getConfig`](https://developers.mattermost.com/extend/plugins/server/reference/#API.GetConfig) API call and can also make changes as needed with [`saveConfig`](https://developers.mattermost.com/extend/plugins/server/reference/#API.SaveConfig).
```
Comment (justinegeffen) [inline site/content/extend/plugins/best-practices.md]:
```suggestion
These settings are stored within the server configuration under [`Plugins`] indexed by plugin ids. The plugin's server code can access their current configuration calling the [`getConfig`](https://developers.mattermost.com/extend/plugins/server/reference/#API.GetConfig) API call and can also make changes as needed with [`saveConfig`](https://developers.mattermost.com/extend/plugins/server/reference/#API.SaveConfig).
```

File: site/content/extend/plugins/best-practices.md [hunk 4]
```diff
@@ -6,11 +6,15 @@ weight: 50
...
+
+## How can I define my own plugin setting type?
+
+A plugin could define their own type of setting with a corresponding custom user interface that will be displayed in the system console following the process below.
```
Comment (justinegeffen) [inline site/content/extend/plugins/best-practices.md]:
```suggestion
A plugin could define its own type of setting with a corresponding custom user interface that will be displayed in the System Console following the process below. 
```

File: site/content/extend/plugins/best-practices.md [hunk 5]
```diff
@@ -6,11 +6,15 @@ weight: 50
...
 
-Additionally, a plugin could define their own type of setting with a corresponding custom user interface that will be displayed in the system console following the process below. 
+These settings are stored within the server configuration under [`Plugins`] indexed by plugin ids. The plugin's server code can access their current configuration calling the [`getConfig`](https://developers.mattermost.com/extend/plugins/server/reference/#API.GetConfig) API call and can also make changes as needed with [`saveConfig`](https://developers.mattermost.com/extend/plugins/server/reference/#API.SaveConfig).
+
+## How can I define my own plugin setting type?
+
+A plugin could define its own type of setting with a corresponding custom user interface that will be displayed in the System Console following the process below.
```
Comment (lieut-data) [inline site/content/extend/plugins/best-practices.md]: nit: both of these section headers have "I" in the title, but then we don't go on to say "you" anywhere. In particular, I stumbled over the subject change here, almost equating "I" with "plugin".

Comment (marianunez) [reply site/content/extend/plugins/best-practices.md]: True, maybe I was just following the "theme" of the original question that was in this section [How can I review the entire code base of a plugin?](https://developers.mattermost.com/extend/plugins/best-practices/#how-can-i-review-the-entire-code-base-of-a-plugin)

Do you think it's more proper to rephrase to maybe something like
```suggestion
How can a plugin define its own setting type?
```

Comment (lieut-data) [reply site/content/extend/plugins/best-practices.md]: I'd be fine with that phrasing :)

Comment (marianunez) [reply site/content/extend/plugins/best-practices.md]: Done!

**Comment by marianunez** [comment]
@aaronrothschild gentle ping

**Pull request merged**